app/models/elastic_search.rb

```diff
@@ -0,0 +1,157 @@
+  def initialize(_index = nil)
+    @es_config = MlnConfigurationController.new.elasticsearch_config('teachersets')
+    arguments = {
+      host: @es_config['host'],
+      transport_options: {
+        request: { open_timeout: @es_config['connect_timeout'] },
+        headers: { content_type: 'application/json' }
+      }
+    }
+    @client = Elasticsearch::Client.new(arguments)
+    @current_file = File.basename(__FILE__)
+    @index = @es_config['index'] || 'teacherset'
+    @type = @es_config['type'] || 'teacherset'
+  end
+
+
+  def index_doc_count
+    response = @client.perform_request 'GET', @index + '/_count'
+    total_docs = response.body['count']
+    total_docs
+  end
+
+
+  def create_document(id, body)
+    @client.create index: @index, type: @type, id: id, body: body
+  end
+
+
+  def delete_document_by_id(id)
+    response = @client.delete index: @index, type: @type, id: id
+    LogWrapper.log('DEBUG', {'message' => "ES document successfully deleted. Id: #{id}", 
+                             'method' => 'delete_document_by_id'})
+
+    response
+  end
+
+
+  def partial_search_by_query(keyword, from, size)
+    query = {"from": from, "size": size, "query": {"multi_match": {"query": keyword, "fields": %w[title description contents]}}}
+    search_by_query(query)
+  end
+
+
+  def fuzzy_search_by_query(keyword, from, size, fuzzy_val)
+    query = {"from": from, "size": size, "query": {"multi_match": 
+      {"query": keyword, "fields": %w[title description contents], "fuzziness": fuzzy_val}}}
+    search_by_query(query)
+  end
+
+
+  def teacher_sets_params(params)
+    keyword = params["keyword"]
+    g_begin = params["grade_begin"]
+    g_end = params["grade_end"]
+    language = params["language"]
+    set_type = params['set type']
+    availability = params['availability']
+    area_of_study = params['area of study']
+    subjects = params['subjects']
+    [keyword, g_begin, g_end, language, set_type, availability, area_of_study, subjects]
+  end
+
+
+  def get_teacher_sets_from_es(params)
+    page = params["page"].present? ? params["page"].to_i - 1 : 0
+    size = 20
+    from = page.to_i * size.to_i
+    query = teacher_sets_query_based_on_filters(params)
+    query[:from] = from
+    query[:size] = size
+    query[:sort] = [{"_score": "desc", "availability.raw": "asc", "created_at": "desc", "_id": "asc"}]
+
+    results = search_by_query(query)
+
+    if !results[:hits].present? && params["keyword"].present? && query[:query][:bool][:must].present?
+      if query[:query][:bool][:must][0][:multi_match].present?
+        query[:query][:bool][:must][0][:multi_match][:fuzziness] = 1
+        results = search_by_query(query)
+      end
+    end
+    results
+  end
+
+
+  def teacher_sets_query_based_on_filters(params)
+    keyword, g_begin, g_end, language, set_type, availability, area_of_study, subjects = teacher_sets_params(params)
+    query = {:query => {:bool => {:must => []}}}
+
+    if keyword.present?
```

Comment:
I feel there are a number of important logic blocks in this method.  I'd like to see one-liner comments, that would help a future maintainer scan the code.